presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -822,27 +849,33 @@ public void dropColumn(HiveIdentity identity, String databaseName, String tableN
     public void addPartitions(HiveIdentity identity, String databaseName, String tableName, List<PartitionWithStatistics> partitions)
     {
         try {
-            List<List<PartitionWithStatistics>> batchedPartitions = Lists.partition(partitions, BATCH_CREATE_PARTITION_MAX_PAGE_SIZE);
-            List<Future<BatchCreatePartitionResult>> futures = new ArrayList<>();
-
-            for (List<PartitionWithStatistics> partitionBatch : batchedPartitions) {
-                List<PartitionInput> partitionInputs = partitionBatch.stream().map(GlueInputConverter::convertPartition).collect(toList());
-                futures.add(glueClient.batchCreatePartitionAsync(new BatchCreatePartitionRequest()
-                        .withCatalogId(catalogId)
-                        .withDatabaseName(databaseName)
-                        .withTableName(tableName)
-                        .withPartitionInputList(partitionInputs)));
-            }
-
-            for (Future<BatchCreatePartitionResult> future : futures) {
-                BatchCreatePartitionResult result = future.get();
-                propagatePartitionErrorToPrestoException(databaseName, tableName, result.getErrors());
-            }
-        }
-        catch (AmazonServiceException | InterruptedException | ExecutionException e) {
-            if (e instanceof InterruptedException) {
-                Thread.currentThread().interrupt();
-            }
+            stats.getAddPartitions().wrap(() -> {
+                List<List<PartitionWithStatistics>> batchedPartitions = Lists.partition(partitions, BATCH_CREATE_PARTITION_MAX_PAGE_SIZE);
+                List<Future<BatchCreatePartitionResult>> futures = new ArrayList<>();
+
+                for (List<PartitionWithStatistics> partitionBatch : batchedPartitions) {
+                    List<PartitionInput> partitionInputs = partitionBatch.stream().map(GlueInputConverter::convertPartition).collect(toList());
+                    futures.add(glueClient.batchCreatePartitionAsync(new BatchCreatePartitionRequest()
+                            .withCatalogId(catalogId)
+                            .withDatabaseName(databaseName)
+                            .withTableName(tableName)
+                            .withPartitionInputList(partitionInputs)));
+                }
+
+                for (Future<BatchCreatePartitionResult> future : futures) {
+                    try {
+                        BatchCreatePartitionResult result = future.get();
+                        propagatePartitionErrorToPrestoException(databaseName, tableName, result.getErrors());
+                    }
+                    catch (InterruptedException e) {
```

Comment:
Add `throw new PrestoException(HIVE_METASTORE_ERROR, e);` to catch clause (after re-interrupting current thread)